packages/gatsby/cache-dir/loader.js

```diff
@@ -315,6 +315,16 @@ export class BaseLoader {
     return page && page.notFound === true
   }
 
+  isPageExist(rawPath) {
+    const pagePath = findPath(rawPath)
+    const page = this.pageDb.get(pagePath)
+    return (
+      page &&
+      stripSurroundingSlashes(page.payload.page.path) ===
+        stripSurroundingSlashes(rawPath)
+    )
+  }
+
```

Comment:
Only thing here is that this method is used only in development, but it does get included in production bundles - we want to avoid cases like that. I'll go ahead and make some adjustments to minimize that

Reply:
handled